keps/sig-node/4381-dra-numeric-parameters/README.md

```diff
@@ -0,0 +1,1501 @@
+apiserver by updating a `PodSchedulingContext` object, ultimately leading to the
+allocation of all pending claims and the pod being scheduled onto a node.
+
+This approach poses a problem for the [Cluster
+Autoscaler](https://github.com/kubernetes/autoscaler) (CA) or for any higher
+level controller that needs to make decisions for a group of pods (e.g. a job
+scheduler). It cannot simulate the effect of allocating or deallocating
+claims over time. Only the third-party DRA
+drivers have the information available to do this.
+
+"Semantic parameters" is an extension to DRA that addresses this problem by
+making claim parameters less opaque. Instead of managing the semantics of all
+claim parameters themselves, drivers now manage resources and describe them
+using a specific "semantic model" pre-defined by Kubernetes. This allows
+components aware of this "semantic model" to make decisions about these
+resources without outsourcing them to some third-party controller. For example,
+the scheduler is now able to allocate claims rapidly, without back-and-forth
+communication with DRA drivers.
+
+At a high-level, this extension takes the following form:
+
+* DRA drivers publish their available resources in the form of a
+  `NodeResourceSlice` object on a node-by-node basis according to one of the
+  builtin "semantic models" known to Kubernetes. This object is stored in the
+  API server and available to the scheduler (or Cluster Autoscaler) to query
+  when a resource request comes in later on.
+
+* When a user wants to consume a resource, they create a `ResourceClaim`,
+  which, in turn, references a claim parameters object. This object defines how
+  many resources are needed and which capabilities they must have. Typically, it
+  is defined using a vendor-specific type which might also support configuration
+  parameters (i.e. parameters that are *not* needed for allocation but *are*
+  needed for configuration).
+
+* With such a claim in place, DRA drivers "resolve" the contents of any
+  vendor-specific claim parameters into a canonical form (i.e. a generic
+  `ResourceClaimParameters` object in the `resource.k8s.io` API group) which
+  the scheduler (or Cluster Autoscaler) can evaluate against the
+  `NodeResourceSlice` of any candidate nodes without knowing exactly what is
+  being requested. They then use this information to help decide which node to
+  schedule a pod on (as well as allocate resources from its `NodeResourceSlice`
+  in the process).
+
+* Once a node is chosen and a workload is sent to it, DRA drivers are
+  responsible for injecting any allocated resources into the Pod, according to
+  the resource choices made by the scheduler. This includes applying any
+  configuration information attached to the vendor-specific claim parameters
+  object used in the request.
+
+In this KEP, we define a single "semantic model" called
+`PartitionableResources`, and use it throughout the discussion to help describe
```

Comment:
I have one problem with "partitionable": I can't spell it reliably. Half the time I write "partionable" instead. It's also not a typo that is easy to spot.

If someone has a suggestion for a name that doesn't have this problem, then I am all ears. We don't need to finalize that before KEP freeze, but definitely before the 1.30 code freeze because the name then becomes part of the API.